### PR TITLE
Update Dockerfile for xdp and cross compile

### PR DESCRIPTION
**Pull request by ami-GS**
## Description

Docker image for build installs XDP dependencies from Ubuntu24.04 repository.
Ubuntu22.04's non XDP test got affected by the mismatched libc version even though it doesn't actually build/use XDP
This change makes multistage build for xdp (ubuntu22.04 with xdp) and cross-compile (others)

image

## Testing

See automation

## Documentation

N/A

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -135,7 +135,7 @@ jobs:
       run: |
         sudo chmod -R 777 artifacts
     - name: Prepare Machine
-      run: scripts/prepare-machine.ps1 -Tls ${{ matrix.vec.tls }} -ForTest ${{ matrix.vec.xdp == '-UseXdp' && '-UseXdp -ForceXdpInstall' || '' }}
+      run: scripts/prepare-machine.ps1 -Tls ${{ matrix.vec.tls }} -ForTest ${{ matri.vec.xdp == '-UseXdp' && '-UseXdp -ForceXdpInstall' || '' }}
```
Comment (nibanks) [inline .github/workflows/test.yml:138]: Typo

File: .azure/OneBranch.Official.yml [hunk 2]
```diff
@@ -76,18 +76,15 @@ extends:
         image: 'cdpxb7b51c2f738e43e48f7605d9a8e5f6d700.azurecr.io/b7b51c2f-738e-43e4-8f76-05d9a8e5f6d7/official/msquicbuild:${{ parameters.WindowsContainerImage2DockerTag }}'
         type: Windows
       - container: linux_build_container # Default container
-        image: 'ghcr.io/microsoft/msquic/linux-build-xcomp:ubuntu-20.04-x86_64'
+        image: 'ghcr.io/microsoft/msquic/linux-build-xcomp:ubuntu-20.04-cross'
         type: Linux
-      - container: ubuntu_2004_x86_64
-        image: 'ghcr.io/microsoft/msquic/linux-build-xcomp:ubuntu-20.04-x86_64'
+      - container: ubuntu_2004_cross
```
Comment (nibanks) [inline .azure/OneBranch.Official.yml:81]: This is the same as the default, so should we merge them?

**Review by nibanks — DISMISSED** [review]

File: .azure/OneBranch.Official.yml [hunk 3]
```diff
@@ -75,9 +75,6 @@ extends:
       - container: windows_build_container2
         image: 'cdpxb7b51c2f738e43e48f7605d9a8e5f6d700.azurecr.io/b7b51c2f-738e-43e4-8f76-05d9a8e5f6d7/official/msquicbuild:${{ parameters.WindowsContainerImage2DockerTag }}'
         type: Windows
-      - container: linux_build_container # Default container
```
Comment (nibanks) [inline .azure/OneBranch.Official.yml:78]: You must specify the default container, with the name `linux_build_container`. Other onebranch stuff depends on it.

**Review by nibanks — APPROVED** [review]

**Comment by ami-GS** [comment]
This is ready to go. error seems to be known issue?

**Review by nibanks — APPROVED** [review]

**Pull request merged**